rust workload failing on debian12 after recent upgrade

### Description of the problem

In Debian 12, rustc v1.63 can be installed from apt, but some of the packages have been upgraded and dependencies are compatible with rustc v1.70.0 and causing build failures.



### Steps to reproduce

cd CI-Examples/rust
make distclean && make


### Expected results

make should be successful and rust workload should be able to run with gramine
```
make
cargo build --release
gramine-manifest \
	-Dlog_level=error \
	-Darch_libdir=/lib/x86_64-linux-gnu \
	-Dself_exe=target/release/rust-hyper-http-server \
	rust-hyper-http-server.manifest.template rust-hyper-http-server.manifest
    Updating crates.io index
 Downloading crates ...
  Downloaded equivalent v1.0.1
  Downloaded want v0.3.1
  Downloaded try-lock v0.2.5
  Downloaded tower-service v0.3.2
  Downloaded cfg-if v1.0.0
  Downloaded autocfg v1.3.0
  Downloaded futures-sink v0.3.30
  Downloaded futures-task v0.3.30
  Downloaded futures-core v0.3.30
  Downloaded fnv v1.0.7
  Downloaded num_cpus v1.16.0
  Downloaded http-body v0.4.6
  Downloaded signal-hook-registry v1.4.2
  Downloaded httpdate v1.0.3
  Downloaded pin-utils v0.1.0
  Downloaded slab v0.4.9
  Downloaded scopeguard v1.2.0
  Downloaded futures-channel v0.3.30
  Downloaded itoa v1.0.11
  Downloaded tokio-macros v2.3.0
  Downloaded lock_api v0.4.12
  Downloaded quote v1.0.36
  Downloaded smallvec v1.13.2
  Downloaded pin-project-lite v0.2.14
  Downloaded parking_lot_core v0.9.10
  Downloaded once_cell v1.19.0
  Downloaded parking_lot v0.12.3
  Downloaded httparse v1.9.4
  Downloaded proc-macro2 v1.0.86
  Downloaded bytes v1.6.1
  Downloaded socket2 v0.5.7
  Downloaded unicode-ident v1.0.12
  Downloaded tracing-core v0.1.32
  Downloaded tracing v0.1.40
  Downloaded indexmap v2.2.6
  Downloaded http v0.2.12
  Downloaded mio v0.8.11
  Downloaded hashbrown v0.14.5
  Downloaded tokio-util v0.7.11
  Downloaded futures-util v0.3.30
  Downloaded h2 v0.3.26
  Downloaded hyper v0.14.30
  Downloaded syn v2.0.72
  Downloaded libc v0.2.155
  Downloaded tokio v1.38.1
   Compiling libc v0.2.155
   Compiling autocfg v1.3.0
   Compiling proc-macro2 v1.0.86
   Compiling unicode-ident v1.0.12
   Compiling pin-project-lite v0.2.14
   Compiling parking_lot_core v0.9.10
   Compiling cfg-if v1.0.0
   Compiling bytes v1.6.1
   Compiling smallvec v1.13.2
   Compiling scopeguard v1.2.0
   Compiling futures-core v0.3.30
   Compiling fnv v1.0.7
   Compiling once_cell v1.19.0
   Compiling itoa v1.0.11
   Compiling hashbrown v0.14.5
   Compiling tracing-core v0.1.32
   Compiling futures-sink v0.3.30
   Compiling pin-utils v0.1.0
   Compiling equivalent v1.0.1
   Compiling futures-task v0.3.30
   Compiling httparse v1.9.4
   Compiling http v0.2.12
   Compiling lock_api v0.4.12
   Compiling quote v1.0.36
   Compiling slab v0.4.9
   Compiling indexmap v2.2.6
   Compiling syn v2.0.72
   Compiling mio v0.8.11
   Compiling socket2 v0.5.7
   Compiling parking_lot v0.12.3
   Compiling num_cpus v1.16.0
   Compiling signal-hook-registry v1.4.2
   Compiling tracing v0.1.40
   Compiling futures-util v0.3.30
   Compiling try-lock v0.2.5
   Compiling want v0.3.1
   Compiling http-body v0.4.6
   Compiling futures-channel v0.3.30
   Compiling httpdate v1.0.3
   Compiling tower-service v0.3.2
   Compiling tokio-macros v2.3.0
   Compiling tokio v1.38.1
   Compiling tokio-util v0.7.11
   Compiling h2 v0.3.26
   Compiling hyper v0.14.30
   Compiling rust-hyper-http-server v0.1.0 (/home/intel/jenkins/workspace/local_ci_graphene_sgx_debian_12_6.1/gramine/CI-Examples/rust)
    Finished release [optimized] target(s) in 2m 45s
```

### Actual results


**_package `tokio v1.39.1` cannot be built because it requires rustc 1.70 or newer_**
```
make
cargo build --release
gramine-manifest \
	-Dlog_level=error \
	-Darch_libdir=/lib/x86_64-linux-gnu \
	-Dself_exe=target/release/rust-hyper-http-server \
	rust-hyper-http-server.manifest.template rust-hyper-http-server.manifest
    Updating crates.io index
 Downloading crates ...
  Downloaded futures-sink v0.3.30
  Downloaded futures-channel v0.3.30
  Downloaded equivalent v1.0.1
  Downloaded tower-service v0.3.2
  Downloaded want v0.3.1
  Downloaded try-lock v0.2.5
  Downloaded autocfg v1.3.0
  Downloaded scopeguard v1.2.0
  Downloaded httpdate v1.0.3
  Downloaded http-body v0.4.6
  Downloaded futures-task v0.3.30
  Downloaded fnv v1.0.7
  Downloaded pin-utils v0.1.0
  Downloaded futures-core v0.3.30
  Downloaded cfg-if v1.0.0
  Downloaded slab v0.4.9
  Downloaded pin-project-lite v0.2.14
  Downloaded signal-hook-registry v1.4.2
  Downloaded itoa v1.0.11
  Downloaded smallvec v1.13.2
  Downloaded tokio-macros v2.4.0
  Downloaded lock_api v0.4.12
  Downloaded parking_lot_core v0.9.10
  Downloaded httparse v1.9.4
  Downloaded bytes v1.6.1
  Downloaded once_cell v1.19.0
  Downloaded proc-macro2 v1.0.86
  Downloaded parking_lot v0.12.3
  Downloaded unicode-ident v1.0.12
  Downloaded quote v1.0.36
  Downloaded socket2 v0.5.7
  Downloaded tracing-core v0.1.32
  Downloaded indexmap v2.2.6
  Downloaded tracing v0.1.40
  Downloaded mio v1.0.1
  Downloaded http v0.2.12
  Downloaded tokio-util v0.7.11
  Downloaded hashbrown v0.14.5
  Downloaded futures-util v0.3.30
  Downloaded h2 v0.3.26
  Downloaded hyper v0.14.30
  Downloaded syn v2.0.72
  Downloaded libc v0.2.155
  Downloaded tokio v1.39.1
error: package `tokio v1.39.1` cannot be built because it requires rustc 1.70 or newer, while the currently active rustc version is 1.63.0
Either upgrade to rustc 1.70 or newer, or use
cargo update -p tokio@1.39.1 --precise ver
where `ver` is the latest version of `tokio` supporting rustc 1.63.0
make: *** [Makefile:26: target/release/rust-hyper-http-server] Error 101
```

### Gramine commit hash

139f6423584322312fbac5f1c1d66731f8d4949e

--- a/CI-Examples/rust/README.md
+++ b/CI-Examples/rust/README.md
@@ -12,6 +12,11 @@ The bind address and port are hardcoded in `src/main.rs`.
 NOTE: The "benchmark-http.sh" script uses the wrk benchmark (wrk2) under the
 hood. Please refer to https://github.com/giltene/wrk2.
 
+NOTE: The Tokio crate is known to require sufficiently new Rust compiler
+(`rustc`). Such new version may not be installed by the OS distro Rust package.
+In such case, please install the latest stable Rust compiler manually, e.g., via
+https://rustup.rs/.
+
 ```sh
 # build the program and the final manifest
 make SGX=1

